foreshadow/cleaners/data_cleaner.py

```diff
@@ -47,15 +47,37 @@ def get_mapping(self, X):
                 ]
                 for c in X
             ],
-            X=X,
+            cols=X.columns,
```

Comment:
This returns an index object, that is iterable, not a list. As long as that is okay we should be good here.

Reply:
Good catch. My tests were still passing when I used this, though I will change to checking for __iter__ instead which should be the best catch-all for iters.

Reply:
Actually looked into this and I think the current checks suffice.

Reply:
issue created.